File: ProtonedMusic-Client/src/app/Models/UserModel.ts
import { AddonRoles } from './AddonRole';
import { NewsModel } from "./NewsModel";

export interface User {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  phoneNumber: number;
  address: string;
  city: string;
  postal: number;
  country: string;
  pictureFile: File | null;
  profilePicturePath: string;
  role?: string;
  addonRoles?: string;
  token?: string;
  newsLikes: NewsModel[];
  newsIds: number[];
}

export function resetUser() {
  return { 
    id: 0, 
    firstName: '', 
    lastName: '', 
    email: '', 
    password: '', 
    phoneNumber: 0, 
    address: '', 
    city: '', 
    postal: 0, 
    country: '',
    pictureFile: null,
    profilePicturePath: '',
    newsLikes: [],
    newsIds: []
};
}

File: ProtonedMusic.Utility/Models/UserModel.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ProtonedMusic.Utility.Models
{
    public class UserModel
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string FirstName { get; set; } = string.Empty;
        [Required]
        public string LastName { get; set; } = string.Empty;

        [Required]
        public string Email { get; set; } = string.Empty;

        [Required]
        public string Password { get; set; }

        [Required]
        public Role Role { get; set; }

        [Required]
        public string PhoneNumber { get; set; } = string.Empty;
        [Required]
        public string Address { get; set; } = string.Empty;
        [Required]
        public string City { get; set; } = string.Empty;
        [Required]
        public string Postal { get; set; } = string.Empty;
        [Required]
        public string Country { get; set; } = string.Empty;

        public string? RefreshToken { get; set; }
        public DateTime RefreshTokenExpiryTime { get; set; }
    }
}
